### PR TITLE
[Program: GCI] Create doc strings for each API

### LINKED ISSUE #404: Create doc strings for Mentorship Relation API to improve Swagger documentation
## Description

As a developer that uses the Swagger UI,
I need good API documentation on Swagger,
so that I can understand how to properly use the backend API.

The Mentorship Relation API resources don't have enough doc strings to reflect their usage or purpose on Swagger UI. By adding doc strings under the respective REST methods, you'll be updating the Swagger UI at the same time. E.g.: POST /login has a full description on Swagger.

Create doc strings for every resource/endpoint of Mentorship Relation API. **Hint:** the file where you might have to do these changes is [app/api/resources/mentorship_relation.py](https://github.com/systers/mentorship-backend/blob/develop/app/api/resources/mentorship_relation.py) file.
Be mindful to explain important parts of the logic behind the APIs, for example if an endpoint is only available to the creator of a request.

### Update [Required]
- [ ] Improve doc strings for every resource/endpoint of Mentorship Relation API. 

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation
2 hours

### PR DESCRIPTION
**Pull request by ilyamaier**
### Description
- Added doc strings for each API in order to improve Swagger documentation

I provided screenshots for some APIs as it would be too long for including every single API. 

Fixes #404 

### Type of Change:
- Code
- Documentation

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
The changes were tested using Swagger UI.

### Screenshots:

Screenshot (114)

Screenshot (115)

Screenshot (116)

Screenshot (117)

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have made corresponding changes to the documentation

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/api/resources/admin.py [hunk 1]
```diff
@@ -20,7 +20,9 @@ class AssignNewUserAdmin(Resource):
     @admin_ns.expect(auth_header_parser, assign_and_revoke_user_admin_request_body, validate=True)
     def post(cls):
         """
-        Assigns a User as a new Admin.
+        Assigns a user as a new admin
+        The user who sent this request must be an admin and the assigned user
+        must exist and do not be an admin.
```
Comment (isabelcosta) [inline app/api/resources/admin.py]:
```suggestion
        must exist and not be an admin.
```

**Review by isabelcosta — CHANGES_REQUESTED** [review]
Hey @IlyaMaier can you please remove the changes in the admin and user API files, since the issue and GCI task you are solving is only referring to the mentorship relation API. We tend to keep PRs changes in the scope of the issue they are solving.

**Comment by ilyamaier** [comment]
> Hey @IlyaMaier can you please remove the changes in the admin and user API files, since the issue and GCI task you are solving is only referring to the mentorship relation API. We tend to keep PRs changes in the scope of the issue they are solving.

@isabelcosta I've done the required changes and updated the description :)

File: app/api/resources/mentorship_relation.py [hunk 2]
```diff
@@ -29,7 +29,9 @@ class SendRequest(Resource):
     @mentorship_relation_ns.response(400, 'Validation error.')
     def post(cls):
         """
-        Creates a new mentorship relation request.
+        Creates a new mentorship relation request
+        Both users must be available and the duration of th relationship must
```
Comment (sara-02) [inline app/api/resources/mentorship_relation.py]: `the`

**Review by sara-02 — CHANGES_REQUESTED** [review]
Missed a typo

File: app/api/resources/mentorship_relation.py (hunk 2 above)
Comment (ilyamaier) [reply app/api/resources/mentorship_relation.py]: oh yes! Thank you! Will change it rn. 😅

**Comment by sara-02** [comment]
Also, what about the inconsistency of the periods? Any thoughts there?

**Comment by ilyamaier** [comment]
> Also, what about the inconsistency of the periods? Any thoughts there?

@sara-02 What exactly do you mean? The periods are always at the end of each sentence in the actual description. They are not displayed in the title, therefore are redundant

File: app/api/resources/mentorship_relation.py (hunk 2 above)
Comment (ilyamaier) [reply app/api/resources/mentorship_relation.py]: have fixed that :)

**Comment by ilyamaier** [comment]
@sara-02 I thought about it one more time. You are right. I mean if the periods don't change anything, why not to add them to have a clear structure of all sentences. So I added them :)

**Review by sara-02 — APPROVED** [review]

**Comment by sara-02** [comment]
I will wait for @isabelcosta to add her final comments and approve the PR and task.

**Review by isabelcosta — APPROVED** [review]
Great work @IlyaMaier ! thank you for your contribution :)

**Comment by Anjali05** [comment]
I think I am tagged in this by mistake?

**Comment by ilyamaier** [comment]
> I think I am tagged in this by mistake?

Oh yes! Sorry!

**Comment by isabelcosta** [comment]
Thank you so much for your contribution @IlyaMaier during the GCI contest 🙌 
I'm closing this PR, since we decided to merge PR #405

**Pull request closed without merging**